vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,862 @@
+
+IMWG ordering will be used or if you'd like to provide your own company-specific ordering here you
+could do this as follows:
+
+```{r}
+aval_resp_imwg <- function(arg) {
+  case_match(
+    arg,
+    "NE" ~ 8,
+    "sCR" ~ 7,
+    "CR" ~ 6,
+    "VGPR" ~ 5,
+    "PR" ~ 4,
+    "MR" ~ 3,
+    "SD" ~ 2,
+    "PD" ~ 1,
+    NA ~ NA_real_
+  )
+}
+
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp_imwg(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADT, AVAL, AVALC)
+)
+```
+
+## Derive Confirmed Response Parameter {#covr}
+
+Confirmation of response require two consecutive readings of applicable disease 
+parameters (biochemical analyses). No minimal time interval, but a different sample
+is required for the confirmation assessment. Bone marrow assessments and imaging do not need to be confirmed. 
+
+If `RS` contains unconfirmed response and confirmation is performed at 
+next scheduled visit we can derive Confirmed Response based on 
+response at subsequent visit.
+
+While deriving confirmed response, the two points below should be taken into consideration:
+
+* Patients will continue in the last Confirmed Response category until there is confirmation of progression or improvement to a higher response status. 
+* Until any response category has been confirmed, Confirmed Response = stable disease (`SD`). 
+
+Let's define a function `aval_resp_conf` that maps numerical values to the responses, so that `PD` is prioritized and concept of "higher response status" is understandable:
+
+```{r}
+aval_resp_conf <- function(arg) {
+  case_match(
+    arg,
+    "PD" ~ 8,
+    "sCR" ~ 7,
+    "CR" ~ 6,
+    "VGPR" ~ 5,
+    "PR" ~ 4,
+    "MR" ~ 3,
+    "SD" ~ 2,
+    "NE" ~ 1,
+    NA ~ NA_real_
+  )
+}
+```
+
+Below table provides a summary of the Confirmed Response status calculation at 
+each time point. The maximum refers to numeric values previously defined in `aval_resp_conf` function.
+
+```{r echo=FALSE}
+list_resp <- tribble(
+  ~"Response at 1st time point", ~"Response at 2nd time point",
+  ~"Confirmed response at 1st time point",
+  "sCR", "sCR", "sCR",
+  "CR", "sCR/CR", "max(CR, last confirmed status)",
+  "VGPR", "sCR/CR/VGPR", "max(VGPR, last confirmed status)",
+  "PR", "sCR/CR/VGPR/PR", "max(PR, last confirmed status)",
+  "MR", "sCR/CR/VGPR/PR/MR", "max(MR, last confirmed status)",
+  "sCR", "CR", "max(CR, last confirmed status)",
+  "sCR/CR", "VGPR", "max(VGPR, last confirmed status)",
+  "sCR/CR/VGPR", "PR", "max(PR, last confirmed status)",
+  "sCR/CR/VGPR/PR", "MR", "max(MR, last confirmed status)",
+  "sCR/CR/VGPR/PR/MR", "SD/PD/NE/NA", "last confirmed status",
+  "SD", "any", "last confirmed status",
+  "NE", "any", "last confirmed status",
+  "PD reason imaging", "any", "PD",
+  "PD reason serum/urine", "PD/death", "PD",
+  "PD reason serum/urine", "sCR/CR/VGPR/PR/MR/SD/NE/NA", "last confirmed status",
```

Comment:
if its first visit and no last confirmed status, then it should be NE otherwise its last confirmed status..

Reply:
updated with "max(NE, last confirmed status)"